src/track/beatgrid.cpp

```diff
@@ -248,8 +262,8 @@ bool BeatGrid::findPrevNextBeats(double dSamples,
     return true;
 }
 
-
-std::unique_ptr<BeatIterator> BeatGrid::findBeats(double startSample, double stopSample) const {
+std::unique_ptr<BeatIterator> BeatGrid::findBeats(
```

Comment:
Future (after #2512 has been merged): If BeatGrid implements a move constructor there is no need to return a new instance by pointer.

If possible we should use values instead of shared/unique pointers, especially for small, lightweight objects.